modules/ray_tracing/doc/content/source/raytracing/Ray.md

```diff
@@ -0,0 +1,71 @@
+# Ray
+
+A `Ray` is the data structure that represents a single ray in the [ray_tracing/index.md] that is generated and traced by a [RayTracingStudy.md]. It can store data and auxiliary data that can modified along the trace via [RayKernels/index.md] and [RayBCs/index.md].
+
+The discussion of how to use and interact with a `Ray` is summarized into the following sections:
+
+- [Defining a Ray Trajectory](#defining-a-ray-trajectory): How to define a `Ray` trajectory for it to be traced using a [RayTracingStudy.md].
+- [Modifying a Ray Trajectory](#modifying-a-ray-trajectory): How to modify a `Ray` trajectory while it is being traced via [RayKernels/index.md] and [RayBCs/index.md].
+- [Using Ray Data](#using-ray-data): How to best access and use the data stored on the `Ray`.
+- [Getting a Ray](#getting-a-ray): How to obtain a `Ray` to be used for tracing.
+
+Useful member variables avaiable on the `Ray` are:
+
+- `currentPoint()` - The current point of the `Ray`. Before being traced, this is the starting point. While being traced, this is the furthest point that the `Ray` has travelled (during RayKernel execution, this is the end of the segment). After being traced, this is the point where the `Ray` was killed.
```

Comment:
It seems like currentPoint/Elem should only be valid during the tracing and null otherwise. Then have a start/stop Point/Elem methods to get the other information